### PR TITLE
2.13.6

### LINKED ISSUE #520: Please compile version for Scala 2.13.6

### PR DESCRIPTION
**Pull request by saeltz**
Updates to Scala 2.13.6.
Refactors sbt syntax.

Fixes #520.
Inspired by https://github.com/sksamuel/scapegoat/pull/488.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -22,8 +22,8 @@ developers := List(
   )
 )
 
-scalaVersion := "2.13.5"
-crossScalaVersions := Seq("2.11.12", "2.12.11", "2.12.12", "2.13.6", "2.13.3", "2.13.4", "2.13.5")
+scalaVersion := "2.13.6"
+crossScalaVersions := Seq("2.11.12", "2.12.11", "2.12.12", "2.12.13", "2.13.3", "2.13.4", "2.13.5", "2.13.6")
```
Comment (saeltz) [inline build.sbt:26]: Latest 2.12 was updated to 2.13.6 by https://github.com/sksamuel/scapegoat/pull/523. This keeps the order and support for latest 2.12.
We might want to drop 2.12.11.

File: src/main/scala/com/sksamuel/scapegoat/Inspection.scala [hunk 2]
```diff
@@ -109,7 +109,7 @@ final case class InspectionContext(global: Global, feedback: Feedback) {
     }
 
     protected def isArray(tree: Tree): Boolean = tree.tpe.typeSymbol.fullName == "scala.Array"
-    protected def isTraversable(tree: Tree): Boolean = tree.tpe <:< typeOf[Traversable[Any]]
+    protected def isIterable(tree: Tree): Boolean = tree.tpe <:< typeOf[Iterable[Any]]
```
Comment (saeltz) [inline src/main/scala/com/sksamuel/scapegoat/Inspection.scala:112]: Possibly related https://github.com/sksamuel/scapegoat/issues/289 & https://github.com/sksamuel/scapegoat/issues/294.

File: src/main/scala/com/sksamuel/scapegoat/inspections/VariableShadowing.scala [hunk 3]
```diff
@@ -16,17 +15,17 @@ class VariableShadowing
...
   def inspector(context: InspectionContext): Inspector =
     new Inspector(context) {
-      override def postTyperTraverser =
+      override def postTyperTraverser: context.Traverser =
         new context.Traverser {
 
           import context.global._
 
-          private val contexts = new mutable.Stack[ListBuffer[String]]()
+          private var contexts: List[ListBuffer[String]] = Nil
```
Comment (saeltz) [inline src/main/scala/com/sksamuel/scapegoat/inspections/VariableShadowing.scala:23]: A `var List` is better than a `mutable.Stack`, so replacing it. Behavior is kept.

**Comment by mr-git** [comment]
when this might get finished?

**Comment by lxwbr** [comment]
Can this be reviewed any soon?

**Comment by mr-git** [comment]
hi!  when this will get merged and published?

**Comment by jozic** [comment]
@mccartney would you have some time to review/merge this any time soon? 
Please :)

**Comment by mccartney** [comment]
Ack. Thanks for tagging me. I somehow must've missed this PR.

I'll attend to it this week.

**Review by mccartney — APPROVED** [review]
LGTM.
The only comment I have is that we have a policy of keeping two minor versions per major version, so some of them need to be cleaned-up. But I'll do that in a separate PR.

Thank you for the contribution!

**Pull request merged**